tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java

```diff
@@ -38,146 +53,198 @@ public CheckCLDR handleCheck(String path, String fullPath, String value, Options
             return this;
         }
 
-        if (path.contains("/personNames/personName")) {
+        if (path.contains("/personNames")) {
+            XPathParts parts = XPathParts.getFrozenInstance(path);
+            switch(parts.getElement(2)) {
+            //ldml/personNames/initialPattern[@type="initial"]
+            default:
+                // skip to rest of handleCheck
+                break;
 
-            // TODO Mark (maybe not here); validate the nameOrderLocales values and the initialPattern
+            case "nameOrderLocales":
+                //ldml/personNames/nameOrderLocales[@order="givenFirst"]
+                Set<String> orderErrors = null;
+                for (String item : SPLIT_SPACE.split(value)) {
+                    boolean mv = LOCALE_MATCH_VALUE.is(item);
+                    if (!mv) {
+                        if (orderErrors == null) {
+                            orderErrors = new LinkedHashSet<>();
+                        }
+                        orderErrors.add(item);
+                    }
+                }
+                if (orderErrors != null) {
+                    result.add(new CheckStatus().setCause(this)
+                        .setMainType(CheckStatus.errorType)
+                        .setSubtype(Subtype.invalidPlaceHolder)
+                        .setMessage("Invalid locales: \"" + JOIN_SPACE.join(orderErrors) + "\""));
+                }
+                return this;
+
+            case "sampleName":
+                //ldml/personNames/sampleName[@item="informal"]/nameField[@type="surname"]
+                if (value.equals("∅∅∅")) {
```

Comment:
why this? (not blocking, just curious)

Reply:
Right now it raises an error 
1. on given if it is empty or 
2. on surname if it is empty while surname2 isn't

So in both cases the one if flags is empty. I could structure it a bit different if you think I should also flag surname2 in case 2.

Reply:
Oh right, I get it now, thanks. Fine as is.